subprojects/core/src/integTest/groovy/org/gradle/api/file/FileCollectionSymlinkIntegrationTest.groovy

```diff
@@ -65,6 +66,74 @@ class FileCollectionSymlinkIntegrationTest extends AbstractIntegrationSpec {
         "project.objects.fileCollection()"   | "project.objects.fileCollection().from(file, symlink, symlinked)"
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/1365")
+    def "detect changes to broken symlink outputs"() {
+        def root = file("root").createDir()
+        def target = file("target")
+        def link = root.file("link")
+
+        buildFile << """
+            import java.nio.file.*
+            class ProducesLink extends DefaultTask {
+                @OutputDirectory File outputDirectory // TODO outputfile?
+    
+                @TaskAction execute() {
+                    def link = Paths.get('${link}')
+                    Files.deleteIfExists(link);
+                    Files.createSymbolicLink(link, Paths.get('${target}'));
```

Comment:
Don't we have a problem with masking absolute paths in `${target}`. How about we use

```suggestion
                    Files.createSymbolicLink(link, Paths.get(project.file('${target.name}')));
```

Reply:
Could you please tell me why you prefer the other?
This is for symlinks only, so we don't have to escape it for Windows.
By using absolute paths, we don't have to define the folder structure twice, but the resulting String will indeed be slightly uglier, is that the concern?
```
def link = Paths.get('/Users/paplorinc/gradle/subprojects/core/build/tmp/test files/FileCollectionSymlinkIntegrationTest/detect_changes_to_b...k_outputs/pdgg/root/link'
```

Reply:
Ah, you are right, I didn't realize this doesn't need to work on Windows. My concern is then that if we ever add support for using symlinks on Windows, then this test will break...

Reply:
It will and they will add the escapes for the paths :)
Is that okay with you, it seems simpler to me this way?

Reply:
It is fine, we can leave it as is for now.